### ISSUE TITLE
CompositeProperty sample

### ISSUE THREAD

**Issue by barisakpunar** [KalikoCMS/KalikoCMS.Core#68]
Could you provide a simple compositeproperty sample usage?

**Comment by fschultz** [comment]
Of course, here's a sample similar to the custom feature property in the demo:
```c#
namespace DemoSite.Business.PropertyTypes {
    using KalikoCMS.Attributes;
    using KalikoCMS.PropertyType;

    [PropertyType("CB43C664-8300-49E4-8908-EEA022CAB9D9", "New Feature", "New Feature", EditorControl)]
    public class NewFeatureProperty : CompositeProperty
    {
        [ImageProperty("Image", Width = 400)]
        public ImageProperty Image { get; set; }

        [Property("Header")]
        public StringProperty Header { get; set; }

        [Property("Feature text")]
        public HtmlProperty Description { get; set; }

        [Property("Feature link")]
        public LinkProperty Url { get; set; }

        public override string Preview {
            get { return Header.Preview; }
        }
    }
}
```

One thing that is different for this type is that the editor is inherited from the base class (the **EditorControl** as the 4th parameter in the PropertyType attribute).

All sub properties are declared the same way the would on any page type.

One issue with the current CompositeProperty is that it has problem instantiating the HTML editor while in list mode. Don't really know why since all other property types seems to work well, but I'm working to fix it.

**Comment by fschultz** [comment]
Usage of the NewFeatureProperty specified above would look something like:
```c#
namespace DemoSite.Models.Pages {
    using Business.PropertyTypes;
    using KalikoCMS.Attributes;
    using KalikoCMS.Core;
    using KalikoCMS.PropertyType;

    /// <summary>
    /// This is the type for our start page. Here we'll use our own custom property type; FeatureProperty
    /// </summary>
    [PageType("StartPage", "Start page", PageTypeDescription = "Used for start page", PreviewImage = "/Assets/Images/startpage.png")]
    public class StartPage : CmsPage {
        [Property("Main feature")]
        public virtual NewFeatureProperty MainFeature { get; set; }
}
```

And in front end:

```html
  @Model.CurrentPage.MainFeature.Image.ToHtml()
  <h2>@Model.CurrentPage.MainFeature.Header</h2>
  <p>@Model.CurrentPage.MainFeature.Description</p>
  <a href="@Model.CurrentPage.MainFeature.Url" class="btn btn-primary">Learn more »</a>
```

**Comment by barisakpunar** [comment]
Thank you.I used collection.But editor as you said  a little bit problematic.

**Comment by fschultz** [comment]
I've just committed an update to the 1.1.0 branch that fixes the problem initializing HTML and MarkDown fields when using composite properties in collections. So composite properties should be fully working now.

**Comment by barisakpunar** [comment]
I couldn't make work CompositeProperty (Both collection and single)
Maybe I made mistakes while updating the changes.
:(

**Comment by fschultz** [comment]
Sorry to hear. Do you get any specific error or does it just not initialize correctly?

I've uploaded the latest development branch as pre-release NuGet packages (version1.1.0-preview) to the NuGet-repository at http://nuget.kaliko.com/ if that can be of any help to get things working.

**Comment by barisakpunar** [comment]
I updated nuget. But again not working. Could you provide a zip of your working solution. Because there are many problems with collection and single CompositeProperty.
If it is difficult to send zip I can try to explain the errors.

**Comment by fschultz** [comment]
I will try to send over a working solution as soon as possible.

**Comment by fschultz** [comment]
I've updated the demo project with the new composite property. There seems to be some issues still with editing in the collection modal (a few times the HTML-field didn't expand completely), I hope to fix these issues soon, but the base functionality should work.

You'll find the code in the new branch:
https://github.com/KalikoCMS/DemoSite.Mvc/tree/1.1.0-dev

Or for downloading the complete project:
https://github.com/KalikoCMS/DemoSite.Mvc/archive/1.1.0-dev.zip

Hope this help getting things working.

**Comment by barisakpunar** [comment]
Thank you.
I noticed something. When I use SqlLite as in the Demo project everything works.
But when I change the database server to SQL Server ( normally I use SQL Server) the problems I will mention arises.
First. Single CompositeProperty in Admin look like this
![untitled](https://cloud.githubusercontent.com/assets/1381704/10757081/1fe78d88-7cb1-11e5-9e4d-fa462c63b736.png)
Single CompositeProperty editor doesn't come with any property.
Second . Collection CompositeProperty Editor is proper. But It doesn't save and When I want to add another CompositeProperty it opens Edit Dialog with the previous Property values.
![untitled2](https://cloud.githubusercontent.com/assets/1381704/10757153/a2616ac2-7cb1-11e5-8d1f-219220c38417.png)

**Comment by fschultz** [comment]
This is a really weird error, I also got the same problems when switching provider to SQL Server. I will look into this. Thank you very much for reporting!

**Comment by barisakpunar** [comment]
I think it is not about SQL server. Because When I create a new SqlLite Database.The same problems happen.

**Comment by fschultz** [comment]
Yes, you're right about that. It seems to be an initialization problem on brand new pages where the properties are not set up correctly.

**Comment by fschultz** [comment]
I think I've fixed both issues - both the non-display of composite properties on new pages as well as the collection modal keeping previous property values. 

I've pushed this commit to the 1.1.0-dev branch but haven't gotten around to update the preview NuGet package yet (need to do a bit of testing to ensure that this is fixed).

**Comment by fschultz** [comment]
Uploaded a new batch of NuGet packages with the latest development code (1.1.0-preview2) to http://nuget.kaliko.com/

**Comment by dotnetshadow** [comment]
Great work, just wondering when 1.1.0 looks to be released, also it would be good to mark off in the milestones the items that have already been fixed all issues appear as open so far so it's hard to know what has been updated and what not if possible?

**Comment by fschultz** [comment]
Thanks! I'm almost done with the preview function (issue #56), things have been a bit slow but I hope to be able to roll out a beta pretty soon.

Good point to flag up what's done. I usually tend to keep the issues open until they are merged to the master branch, but that will make it very unclear what's fixed or waiting to be fixed. I've added a new flag ("done") that I've attached to issues that are done but waiting to be released. I hope that helps.
